rewrite-java-test/src/test/java/org/openrewrite/java/ReplaceStringLiteralWithConstantTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class ReplaceStringLiteralWithConstantTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+            .parser(JavaParser.fromJavaVersion()
+                .logCompilationWarningsAndErrors(true)
+                .classpath("guava"));
+    }
+
+    @Test
+    void doNothingIfStringLiteralNotFound() {
+        rewriteRun(
+            spec -> spec.recipe(new ReplaceStringLiteralWithConstant("com.google.common.base.Charsets", "UTF_8", "UTF_8")),
+            java(
+                """
+                  class Test {
+                      void test(Object obj) {
+                          String s = "FooBar";
+                      }
+                  }
+                  """
+            )
+        );
+    }
+
+    @Test
+    void replaceStringLiteralWithConstant() {
+        rewriteRun(
+            spec -> spec.recipe(new ReplaceStringLiteralWithConstant("com.google.common.base.Charsets", "UTF_8", "UTF_8")),
+            java(
+                """
+                  class Test {
+                      Object o = "UTF_8";
+                  }
+                  """,
+                """
+                  import com.google.common.base.Charsets;
```

Comment:
Maybe a static import of the constant would be nice? That way there's typically a minimal change to the line where the literal is used.

Reply:
I feel we should make non-static import only , this way we don't have to worry about conflicts and its code is more readable as well. let me know what you think.

Reply:
Agreed; might complicate the recipe a bit indeed; and we can always tie on another recipe to switch from class imports for constants to direct constant imports, and solve the conflicts there. Thanks again!